src/smf/pfcp-sm.c

```diff
@@ -185,8 +185,19 @@ void smf_pfcp_state_associated(ogs_fsm_t *s, smf_event_t *e)
         xact = e->pfcp_xact;
         ogs_assert(xact);
 
-        if (message->h.seid_presence && message->h.seid != 0)
-            sess = smf_sess_find_by_seid(message->h.seid);
+        if (message->h.seid_presence && message->h.seid != 0) {
+               sess = smf_sess_find_by_seid(message->h.seid);
+        } else if (xact->step == 2) { /* rx no SEID or SEID=0 */
+            /* 3GPP TS 29.244 7.2.2.4.2: we receive SEID=0 under some
+               conditions, such as cause "Session context not found". In those
+               cases, we still want to identify the local session which
+               originated the message, so try harder by using the SEID we used
+               to send the original request: */
+            ogs_pkbuf_t *pkbuf = xact->seq[xact->step-1].pkbuf;
+            ogs_pfcp_header_t *h = (ogs_pfcp_header_t *)pkbuf->data;
+            if (h->seid_presence)
+                sess = smf_sess_find_by_seid(be64toh(h->seid));
```

Comment:
Please @acetcom , review this seid part is correct. IIUC in pkbuf the h->seid is in network byte order, but in message->h.seid above the code path already did the swap to host byte order for us.